Blocksworld/blocksWorldViz.js

```diff
@@ -34,13 +31,38 @@ function visualizeStateInDiv(planVizDiv, plan, finalState, displayWidth) {
    gripper.innerHTML = valueMap.get('handempty') ? gripperOpen : gripperClosed;
    blocksState.appendChild(gripper);
 
-   // mock display block held by the gripper:
-   paintGrippedBlock("blue", blocksState);
+   const onTableKeys = [...valueMap.keys()]
+      .filter(key => key.startsWith("ontable"))
+   console.log(onTableKeys);
 
-   // mock display green block on table
-   paintBlock("green", blocksState, /*tower*/3, /*level*/0);
-   // mock display red sitting on green
-   paintBlock("red", blocksState, /*tower*/3, /*level*/1);
+   let towerIndex = 0;
+   for (let key of valueMap.keys()) {
+      if (key.startsWith("ontable")) {
+         let blockName = key.replace("ontable ", "");
+         paintBlock(blockName, blocksState, towerIndex, 0)
+         let levelIndex = 1;
+         while (!valueMap.get("clear " + blockName)) {
+            for (let specialKey of valueMap.keys())
+               if (specialKey.startsWith("on ") && specialKey.endsWith(blockName)) {
+                  let specialKeyName = specialKey
```

Comment:
The value the `specialKeyName` is not the `key`. It is the _other_ block's name. Shall we call it `otherBlockName`?